src/Microsoft.Health.Fhir.Core/Features/Operations/Import/CancelImportRequestHandler.cs

```diff
@@ -43,19 +45,41 @@ public async Task<CancelImportResponse> Handle(CancelImportRequest request, Canc
                 throw new UnauthorizedFhirActionException();
             }
 
-            JobInfo jobInfo = await _queueClient.GetJobByIdAsync(QueueType.Import, request.JobId, false, cancellationToken);
+            // We need to check all Jobs status
+            IReadOnlyList<JobInfo> jobs = await _queueClient.GetJobByGroupIdAsync(QueueType.Import, request.JobId, false, cancellationToken);
 
-            if (jobInfo == null)
+            if (jobs == null || jobs.Count == 0)
             {
                 throw new ResourceNotFoundException(string.Format(Core.Resources.ImportJobNotFound, request.JobId));
             }
 
-            if (jobInfo.Status == JobManagement.JobStatus.Completed || jobInfo.Status == JobManagement.JobStatus.Cancelled || jobInfo.Status == JobManagement.JobStatus.Failed)
+            var conflict = false;
+            var allComplete = true;
+
+            // Check each job status
+            foreach (var job in jobs)
+            {
+                if (job.Status == JobStatus.Failed || job.Status == JobStatus.Cancelled)
+                {
+                    conflict = true;
+                }
+
+                if (job.Status != JobStatus.Completed)
+                {
+                    allComplete = false;
+                }
+            }
+
+            // If the job is already completed, failed or cancelled, return conflict status.
+            if (conflict || allComplete)
             {
                 throw new OperationFailedException(Core.Resources.ImportOperationCompleted, HttpStatusCode.Conflict);
```

Comment:
I tried to use `return new CancelImportResponse(HttpStatusCode.Conflict);` but looks like it process the request as 500 instead of 400, is it correct given import behavior?